Build failure: dmraid

## Steps To Reproduce

Steps to reproduce the behavior:

1. `nix-build -A dmraid` on master (f8aa427967411ed1843040134f742d68fa400f73)

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/j6ra0kjdjibnavjwk5m5lf616bi23hbc-dmraid-1.0.0.rc16.tar.bz2
source root is dmraid
setting SOURCE_DATE_EPOCH to timestamp 1253140355 of file dmraid/1.0.0.rc16/lib/Makefile.in
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/zda1d78g307zc3qwahnwq5fhaxpl3hn0-hardening-format.patch
patching file 1.0.0.rc16/lib/events/libdmraid-events-isw.c
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./1.0.0.rc16/autoconf/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./1.0.0.rc16/autoconf/config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --disable-static --prefix=/nix/store/75k5rywwbb24z7g463cixdh4qnxisjc0-dmraid-1.0.0.rc16
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking for gawk... gawk
checking for a BSD-compatible install... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/install -c
checking whether ln -s works... yes
checking whether make sets $(MAKE)... yes
checking for ranlib... ranlib
checking for gcc... gcc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables... 
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep
checking for egrep... /nix/store/aap6cq56amx4mzbyxp2wpgsf1kqjcr1f-gnugrep-3.11/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking for unistd.h... (cached) yes
checking malloc.h usability... yes
checking malloc.h presence... yes
checking for malloc.h... yes
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking for off_t... yes
checking for pid_t... yes
checking for size_t... yes
checking for struct stat.st_rdev... yes
checking whether time.h and sys/time.h may both be included... yes
checking whether to enable debugging... 
no
checking whether to enable malloc debugging... 
no
checking whether to disable native metadata logging... 
yes
checking whether to disable testing with mapped devices... 
no
checking whether gcc needs -traditional... no
checking return type of signal handlers... void
checking for vprintf... yes
checking for _doprnt... no
checking for mkdir... yes
checking for rmdir... yes
checking for uname... yes
checking getopt.h usability... yes
checking getopt.h presence... yes
checking for getopt.h... yes
configure: creating ./config.status
config.status: creating include/Makefile
config.status: creating lib/Makefile
config.status: creating man/Makefile
config.status: creating tools/Makefile
config.status: creating tools/version.h
config.status: creating Makefile
config.status: creating make.tmpl
config.status: WARNING:  make.tmpl.in seems to ignore the --datarootdir setting
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash
make -C include
make[1]: Entering directory '/build/dmraid/1.0.0.rc16/include'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/build/dmraid/1.0.0.rc16/include'
make -C lib
make[1]: Entering directory '/build/dmraid/1.0.0.rc16/lib'
gcc -MM -MF activate/activate.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE activate/activate.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE activate/activate.c -o activate/activate.o
gcc -MM -MF activate/devmapper.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE activate/devmapper.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE activate/devmapper.c -o activate/devmapper.o
gcc -MM -MF device/ata.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE device/ata.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE device/ata.c -o device/ata.o
gcc -MM -MF device/partition.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE device/partition.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE device/partition.c -o device/partition.o
gcc -MM -MF device/scan.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE device/scan.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE device/scan.c -o device/scan.o
gcc -MM -MF device/scsi.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE device/scsi.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE device/scsi.c -o device/scsi.o
gcc -MM -MF display/display.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE display/display.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE display/display.c -o display/display.o
gcc -MM -MF format/format.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/format.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/format.c -o format/format.o
gcc -MM -MF locking/locking.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE locking/locking.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE locking/locking.c -o locking/locking.o
gcc -MM -MF log/log.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE log/log.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE log/log.c -o log/log.o
gcc -MM -MF metadata/log_ops.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE metadata/log_ops.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE metadata/log_ops.c -o metadata/log_ops.o
gcc -MM -MF metadata/metadata.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE metadata/metadata.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE metadata/metadata.c -o metadata/metadata.o
metadata/metadata.c: In function 'create_raidset':
metadata/metadata.c:1436:24: warning: comparison between pointer and zero character constant []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-pointer-compare-Wpointer-compare]8;;]
 1436 |         } while (end++ != '\0');
      |                        ^~
metadata/metadata.c:1436:18: note: did you mean to dereference the pointer?
 1436 |         } while (end++ != '\0');
      |                  ^
gcc -MM -MF metadata/reconfig.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE metadata/reconfig.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE metadata/reconfig.c -o metadata/reconfig.o
metadata/reconfig.c: In function 'handle_dev':
metadata/reconfig.c:462:29: warning: comparison between 'enum change_type' and 'enum type' []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wenum-compare-Wenum-compare]8;;]
  462 |                 if (h->type == rs->type)
      |                             ^~
metadata/reconfig.c:464:30: warning: comparison between 'enum change_type' and 'enum type' []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wenum-compare-Wenum-compare]8;;]
  464 |         } while ((++h)->type != t_undef);
      |                              ^~
gcc -MM -MF misc/file.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE misc/file.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE misc/file.c -o misc/file.o
gcc -MM -MF misc/init.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE misc/init.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE misc/init.c -o misc/init.o
gcc -MM -MF misc/lib_context.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE misc/lib_context.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE misc/lib_context.c -o misc/lib_context.o
gcc -MM -MF misc/misc.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE misc/misc.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE misc/misc.c -o misc/misc.o
gcc -MM -MF misc/workaround.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE misc/workaround.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE misc/workaround.c -o misc/workaround.o
gcc -MM -MF mm/dbg_malloc.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE mm/dbg_malloc.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE mm/dbg_malloc.c -o mm/dbg_malloc.o
gcc -MM -MF format/ataraid/asr.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/asr.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/asr.c -o format/ataraid/asr.o
format/ataraid/asr.c: In function 'check_rd':
format/ataraid/asr.c:1034:25: warning: comparison between 'enum type' and 'enum status' []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wenum-compare-Wenum-compare]8;;]
 1034 |         return rd->type != s_broken;
      |                         ^~
gcc -MM -MF format/ataraid/hpt37x.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/hpt37x.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/hpt37x.c -o format/ataraid/hpt37x.o
format/ataraid/hpt37x.c: In function 'hpt37x_log':
format/ataraid/hpt37x.c:356:19: warning: taking address of packed member of 'struct hpt37x' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  356 |         for (el = hpt->errorlog; el < hpt->errorlog + 32; el++) {
      |                   ^~~
gcc -MM -MF format/ataraid/hpt45x.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/hpt45x.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/hpt45x.c -o format/ataraid/hpt45x.o
gcc -MM -MF format/ataraid/isw.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/isw.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/isw.c -o format/ataraid/isw.o
format/ataraid/isw.c: In function '_get_disk':
format/ataraid/isw.c:141:41: warning: taking address of packed member of 'struct isw' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  141 |                 struct isw_disk *disk = isw->disk;
      |                                         ^~~
format/ataraid/isw.c: In function 'name':
format/ataraid/isw.c:203:33: warning: taking address of packed member of 'struct isw' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  203 |         struct isw_disk *disk = isw->disk;
      |                                 ^~~
format/ataraid/isw.c: In function 'get_rs_status':
format/ataraid/isw.c:1127:24: warning: taking address of packed member of 'struct isw' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
 1127 |                 disk = isw->disk;
      |                        ^~~
format/ataraid/isw.c:1107:26: warning: variable 'disk' set but not used []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-unused-but-set-variable-Wunused-but-set-variable]8;;]
 1107 |         struct isw_disk *disk;
      |                          ^~~~
format/ataraid/isw.c: In function 'get_rebuild_state':
format/ataraid/isw.c:1178:32: warning: taking address of packed member of 'struct isw' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
 1178 |                         disk = isw->disk;
      |                                ^~~
format/ataraid/isw.c: In function 'isw_metadata_handler':
format/ataraid/isw.c:1310:24: warning: taking address of packed member of 'struct isw' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
 1310 |                 disk = isw->disk;
      |                        ^~~
format/ataraid/isw.c: In function '_isw_create_first_volume':
format/ataraid/isw.c:1821:16: warning: taking address of packed member of 'struct isw' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
 1821 |         disk = isw->disk;
      |                ^~~
format/ataraid/isw.c: In function '_isw_create_second_volume':
format/ataraid/isw.c:1918:44: warning: taking address of packed member of 'struct isw' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
 1918 |         total_size = _cal_array_size(isw_v1->disk, rs, dev1);
      |                                      ~~~~~~^~~~~~
format/ataraid/isw.c: In function '_isw_log':
format/ataraid/isw.c:2185:28: warning: taking address of packed member of 'struct isw' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
 2185 |         for (d = 0, disk = isw->disk; d < isw->num_disks; d++, disk++) {
      |                            ^~~
format/ataraid/isw.c: In function 'update_metadata':
format/ataraid/isw.c:2655:33: warning: taking address of packed member of 'struct isw' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
 2655 |         struct isw_disk *disk = isw->disk, *new_disk = NULL;
      |                                 ^~~
format/ataraid/isw.c: In function 'isw_remove_dev':
format/ataraid/isw.c:2390:17: warning: '__builtin_strncpy' output truncated before terminating nul copying 6 bytes from a string of the same length []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation]8;;]
 2390 |                 strncpy((char *) isw_tmp->sig + MPB_SIGNATURE_SIZE,
      |                 ^
format/ataraid/isw.c:2395:17: warning: '__builtin_strncpy' output truncated before terminating nul copying 6 bytes from a string of the same length []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation]8;;]
 2395 |                 strncpy((char *) isw_tmp->sig + MPB_SIGNATURE_SIZE,
      |                 ^
format/ataraid/isw.c: In function 'isw_config_dev':
format/ataraid/isw.c:1708:9: warning: '__builtin_strncpy' specified bound 16 equals destination size []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation]8;;]
 1708 |         strncpy((char *) dev2->volume, rs->name, MAX_RAID_SERIAL_LEN);
      |         ^
format/ataraid/isw.c: In function 'isw_create':
format/ataraid/isw.c:1843:9: warning: '__builtin_strncpy' output truncated before terminating nul copying 24 bytes from a string of the same length []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation]8;;]
 1843 |         strncpy((char *) isw->sig, MPB_SIGNATURE, MPB_SIGNATURE_SIZE);
      |         ^
format/ataraid/isw.c:2770:9: warning: '__builtin_strncpy' specified bound 16 equals destination size []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation]8;;]
 2770 |         strncpy((char *) new_disk->serial, dev_info_serial_to_isw(di->serial),
      |         ^
format/ataraid/isw.c:1524:17: warning: '__builtin_strncpy' specified bound 16 equals destination size []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-stringop-truncation-Wstringop-truncation]8;;]
 1524 |                 strncpy((char *) disk[i].serial,
      |                 ^
gcc -MM -MF format/ataraid/jm.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/jm.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/jm.c -o format/ataraid/jm.o
format/ataraid/jm.c: In function 'name':
format/ataraid/jm.c:35:18: warning: 'i' is used uninitialized []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-uninitialized-Wuninitialized]8;;]
   35 |         while (i && isspace(buf[i])) {
      |                  ^
format/ataraid/jm.c:28:16: note: 'i' was declared here
   28 |         size_t i, len;
      |                ^
gcc -MM -MF format/ataraid/lsi.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/lsi.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/lsi.c -o format/ataraid/lsi.o
gcc -MM -MF format/ataraid/nv.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/nv.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/nv.c -o format/ataraid/nv.o
gcc -MM -MF format/ataraid/pdc.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/pdc.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/pdc.c -o format/ataraid/pdc.o
format/ataraid/pdc.c: In function '_pdc_log':
format/ataraid/pdc.c:588:21: warning: taking address of packed member of 'struct pdc' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  588 |         for (disk = pdc->raid.disk, i = 0;
      |                     ^~~
gcc -MM -MF format/ataraid/sil.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/sil.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/sil.c -o format/ataraid/sil.o
gcc -MM -MF format/ataraid/via.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/via.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ataraid/via.c -o format/ataraid/via.o
gcc -MM -MF format/ddf/ddf1.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ddf/ddf1.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ddf/ddf1.c -o format/ddf/ddf1.o
format/ddf/ddf1.c: In function 'check_rd':
format/ddf/ddf1.c:876:25: warning: comparison between 'enum type' and 'enum status' []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wenum-compare-Wenum-compare]8;;]
  876 |         return rd->type != s_broken;
      |                         ^~
gcc -MM -MF format/ddf/ddf1_lib.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ddf/ddf1_lib.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ddf/ddf1_lib.c -o format/ddf/ddf1_lib.o
gcc -MM -MF format/ddf/ddf1_crc.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ddf/ddf1_crc.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ddf/ddf1_crc.c -o format/ddf/ddf1_crc.o
format/ddf/ddf1_crc.c: In function 'crc32_vd':
format/ddf/ddf1_crc.c:100:24: warning: taking address of packed member of 'struct ddf1_config_record' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  100 |                 .crc = &r->crc, \
      |                        ^~~~~~~
format/ddf/ddf1_crc.c:108:1: note: in expansion of macro 'CRC32'
  108 | CRC32(vd, ddf1_config_record, CR);
      | ^~~~~
format/ddf/ddf1_crc.c: In function 'crc32_spare':
format/ddf/ddf1_crc.c:100:24: warning: taking address of packed member of 'struct ddf1_spare_header' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  100 |                 .crc = &r->crc, \
      |                        ^~~~~~~
format/ddf/ddf1_crc.c:109:1: note: in expansion of macro 'CRC32'
  109 | CRC32(spare, ddf1_spare_header, SR);
      | ^~~~~
format/ddf/ddf1_crc.c: In function 'vd_check_crc':
format/ddf/ddf1_crc.c:149:24: warning: taking address of packed member of 'struct ddf1_config_record' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  149 |                 .crc = &r->crc, \
      |                        ^~~~~~~
format/ddf/ddf1_crc.c:156:1: note: in expansion of macro 'CHECK_CRC'
  156 | CHECK_CRC(vd, ddf1_config_record, CR, "VD CFG");
      | ^~~~~~~~~
format/ddf/ddf1_crc.c: In function 'spare_check_crc':
format/ddf/ddf1_crc.c:149:24: warning: taking address of packed member of 'struct ddf1_spare_header' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  149 |                 .crc = &r->crc, \
      |                        ^~~~~~~
format/ddf/ddf1_crc.c:157:1: note: in expansion of macro 'CHECK_CRC'
  157 | CHECK_CRC(spare, ddf1_spare_header, SR, "Spare CFG");
      | ^~~~~~~~~
format/ddf/ddf1_crc.c: In function 'all_crcs':
format/ddf/ddf1_crc.c:182:33: warning: taking address of packed member of 'struct ddf1_header' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  182 |                 {ddf1->primary, &ddf1->primary->crc,
      |                                 ^~~~~~~~~~~~~~~~~~~
format/ddf/ddf1_crc.c:185:35: warning: taking address of packed member of 'struct ddf1_header' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  185 |                 {ddf1->secondary, &ddf1->secondary->crc,
      |                                   ^~~~~~~~~~~~~~~~~~~~~
format/ddf/ddf1_crc.c:188:33: warning: taking address of packed member of 'struct ddf1_adapter' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  188 |                 {ddf1->adapter, &ddf1->adapter->crc,
      |                                 ^~~~~~~~~~~~~~~~~~~
format/ddf/ddf1_crc.c:191:35: warning: taking address of packed member of 'struct ddf1_disk_data' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  191 |                 {ddf1->disk_data, &ddf1->disk_data->crc,
      |                                   ^~~~~~~~~~~~~~~~~~~~~
format/ddf/ddf1_crc.c:194:35: warning: taking address of packed member of 'struct ddf1_phys_drives' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  194 |                 {ddf1->pd_header, &ddf1->pd_header->crc,
      |                                   ^~~~~~~~~~~~~~~~~~~~~
format/ddf/ddf1_crc.c:198:35: warning: taking address of packed member of 'struct ddf1_virt_drives' may result in an unaligned pointer value []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-of-packed-member-Waddress-of-packed-member]8;;]
  198 |                 {ddf1->vd_header, &ddf1->vd_header->crc,
      |                                   ^~~~~~~~~~~~~~~~~~~~~
gcc -MM -MF format/ddf/ddf1_cvt.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ddf/ddf1_cvt.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ddf/ddf1_cvt.c -o format/ddf/ddf1_cvt.o
gcc -MM -MF format/ddf/ddf1_dump.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ddf/ddf1_dump.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/ddf/ddf1_dump.c -o format/ddf/ddf1_dump.o
gcc -MM -MF format/partition/dos.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/partition/dos.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE format/partition/dos.c -o format/partition/dos.o
gcc -MM -MF register/dmreg.d -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE register/dmreg.c; \
gcc -c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE register/dmreg.c -o register/dmreg.o
rm -f ../lib/libdmraid.a
ar r ../lib/libdmraid.a activate/activate.o activate/devmapper.o device/ata.o device/partition.o device/scan.o device/scsi.o display/display.o format/format.o locking/locking.o log/log.o metadata/log_ops.o metadata/metadata.o metadata/reconfig.o misc/file.o misc/init.o misc/lib_context.o misc/misc.o misc/workaround.o mm/dbg_malloc.o format/ataraid/asr.o format/ataraid/hpt37x.o format/ataraid/hpt45x.o format/ataraid/isw.o format/ataraid/jm.o format/ataraid/lsi.o format/ataraid/nv.o format/ataraid/pdc.o format/ataraid/sil.o format/ataraid/via.o format/ddf/ddf1.o format/ddf/ddf1_lib.o format/ddf/ddf1_crc.o format/ddf/ddf1_cvt.o format/ddf/ddf1_dump.o format/partition/dos.o register/dmreg.o
ar: creating ../lib/libdmraid.a
ranlib ../lib/libdmraid.a
gcc -shared -Wl,-soname,libdmraid.so.1.0.0.rc16 \
-Wl,--version-script,.export.sym activate/activate.o activate/devmapper.o device/ata.o device/partition.o device/scan.o device/scsi.o display/display.o format/format.o locking/locking.o log/log.o metadata/log_ops.o metadata/metadata.o metadata/reconfig.o misc/file.o misc/init.o misc/lib_context.o misc/misc.o misc/workaround.o mm/dbg_malloc.o format/ataraid/asr.o format/ataraid/hpt37x.o format/ataraid/hpt45x.o format/ataraid/isw.o format/ataraid/jm.o format/ataraid/lsi.o format/ataraid/nv.o format/ataraid/pdc.o format/ataraid/sil.o format/ataraid/via.o format/ddf/ddf1.o format/ddf/ddf1_lib.o format/ddf/ddf1_crc.o format/ddf/ddf1_cvt.o format/ddf/ddf1_dump.o format/partition/dos.o register/dmreg.o -o ../lib/libdmraid.so -ldevmapper-event
gcc events/libdmraid-events-isw.c -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE  -shared  -Wl,-soname,libdmraid-events-isw.so.1.0.0.rc16 -o ../lib/libdmraid-events-isw.so -ldevmapper -ldevmapper-event ../lib/libdmraid.so
events/libdmraid-events-isw.c: In function '_snprintf_either':
events/libdmraid-events-isw.c:807:35: warning: the comparison will always evaluate as 'true' for the address of 'name' will never be NULL []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Waddress-Waddress]8;;]
  807 |                         dev->name ? dev->name : "",
      |                                   ^
events/libdmraid-events-isw.c:132:14: note: 'name' declared here
  132 |         char name[DEV_NAME];            /* sda, sdb, etc... */
      |              ^~~~
events/libdmraid-events-isw.c: In function 'register_device':
events/libdmraid-events-isw.c:1527:28: warning: passing argument 1 of '__xpg_basename' discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdiscarded-qualifiers-Wdiscarded-qualifiers]8;;]
 1527 |         rs_name = basename(rs_name);
      |                            ^~~~~~~
In file included from events/libdmraid-events-isw.c:78:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/libgen.h:34:36: note: expected 'char *' but argument is of type 'const char *'
   34 | extern char *__xpg_basename (char *__path) __THROW;
      |                              ~~~~~~^~~~~~
events/libdmraid-events-isw.c: In function 'unregister_device':
events/libdmraid-events-isw.c:1588:28: warning: passing argument 1 of '__xpg_basename' discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdiscarded-qualifiers-Wdiscarded-qualifiers]8;;]
 1588 |         rs_name = basename(rs_name);
      |                            ^~~~~~~
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/libgen.h:34:36: note: expected 'char *' but argument is of type 'const char *'
   34 | extern char *__xpg_basename (char *__path) __THROW;
      |                              ~~~~~~^~~~~~
events/libdmraid-events-isw.c: At top level:
events/libdmraid-events-isw.c:110:20: warning: 'syslog_ident' defined but not used []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-unused-variable-Wunused-variable]8;;]
  110 | static const char *syslog_ident = "libdmraid-events";
      |                    ^~~~~~~~~~~~
events/libdmraid-events-isw.c:104:20: warning: 'sys_dm_path' defined but not used []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-unused-variable-Wunused-variable]8;;]
  104 | static const char *sys_dm_path = "/sys/block/dm-";
      |                    ^~~~~~~~~~~
events/libdmraid-events-isw.c: In function '_set_raid_dev_properties':
events/libdmraid-events-isw.c:631:46: warning: '%s' directive writing up to 255 bytes into a region of size 233 []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-overflow-Wformat-overflow=]8;;]
  631 |                         sprintf(path + len, "%s%s%c%s", d_name,
      |                                              ^~
In file included from /nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/stdio.h:970,
                 from events/libdmraid-events-isw.c:72:
In function 'sprintf',
    inlined from '_set_raid_dev_properties' at events/libdmraid-events-isw.c:631:4:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/stdio2.h:30:10: note: '__builtin___sprintf_chk' output 15 or more bytes (assuming 270) into a destination of size 233
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
events/libdmraid-events-isw.c: In function '_set_raid_dev_properties':
events/libdmraid-events-isw.c:631:46: warning: '%s' directive writing up to 255 bytes into a region of size 233 []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-overflow-Wformat-overflow=]8;;]
  631 |                         sprintf(path + len, "%s%s%c%s", d_name,
      |                                              ^~
In function 'sprintf',
    inlined from '_set_raid_dev_properties' at events/libdmraid-events-isw.c:631:4:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/stdio2.h:30:10: note: '__builtin___sprintf_chk' output 15 or more bytes (assuming 270) into a destination of size 233
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
events/libdmraid-events-isw.c: In function '_process_mirror_event':
events/libdmraid-events-isw.c:1236:13: warning: 'log_status_str' may be used uninitialized []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wmaybe-uninitialized-Wmaybe-uninitialized]8;;]
 1236 |             *log_status_str == 'D') {
      |             ^~~~~~~~~~~~~~~
events/libdmraid-events-isw.c:1136:15: note: 'log_status_str' was declared here
 1136 |              *log_status_str, *p, *sync_str;
      |               ^~~~~~~~~~~~~~
events/libdmraid-events-isw.c: In function '_create_raid_set.constprop':
events/libdmraid-events-isw.c:905:26: warning: '%s' directive writing up to 255 bytes into a region of size 245 []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-overflow-Wformat-overflow=]8;;]
  905 |         sprintf(path, "%s%s%s", sys_block_path, dev_name, sys_slaves_dir);
      |                          ^~
In function 'sprintf',
    inlined from '_get_slave_devices' at events/libdmraid-events-isw.c:905:2,
    inlined from '_create_raid_set.constprop' at events/libdmraid-events-isw.c:1006:8:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/stdio2.h:30:10: note: '__builtin___sprintf_chk' output between 19 and 274 bytes into a destination of size 256
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
events/libdmraid-events-isw.c: In function '_create_raid_set.constprop':
events/libdmraid-events-isw.c:914:41: warning: '__builtin___sprintf_chk' may write a terminating nul past the end of the destination []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-overflow-Wformat-overflow=]8;;]
  914 |                 sprintf(path + len, "/%s", d_name);
      |                                         ^
In function 'sprintf',
    inlined from '_get_slave_devices' at events/libdmraid-events-isw.c:914:3,
    inlined from '_create_raid_set.constprop' at events/libdmraid-events-isw.c:1006:8:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/stdio2.h:30:10: note: '__builtin___sprintf_chk' output between 2 and 257 bytes into a destination of size 256
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
events/libdmraid-events-isw.c: In function '_create_raid_set.constprop':
events/libdmraid-events-isw.c:422:26: warning: '%s' directive writing up to 255 bytes into a region of size 245 []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-overflow-Wformat-overflow=]8;;]
  422 |         sprintf(path, "%s%s", sys_block_path, dev_name);
      |                          ^~
In function 'sprintf',
    inlined from '_check_raid_dev_active' at events/libdmraid-events-isw.c:422:2,
    inlined from '_get_slave_devices' at events/libdmraid-events-isw.c:922:3,
    inlined from '_create_raid_set.constprop' at events/libdmraid-events-isw.c:1006:8:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/stdio2.h:30:10: note: '__builtin___sprintf_chk' output between 12 and 267 bytes into a destination of size 256
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
In function '_event_cleanup_and_log',
    inlined from '_process_stripe_event' at events/libdmraid-events-isw.c:1126:2:
events/libdmraid-events-isw.c:534:12: warning: 'args' may be used uninitialized []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wmaybe-uninitialized-Wmaybe-uninitialized]8;;]
  534 |         if (args)
      |            ^
events/libdmraid-events-isw.c: In function '_process_stripe_event':
events/libdmraid-events-isw.c:1071:16: note: 'args' was declared here
 1071 |         char **args, *dev_status_str, *p;
      |                ^~~~
In function '_check_sgpio',
    inlined from 'register_device' at events/libdmraid-events-isw.c:1525:2:
events/libdmraid-events-isw.c:179:17: warning: 'fclose' called on pointer returned from a mismatched allocation function []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wmismatched-dealloc-Wmismatched-dealloc]8;;]
  179 |                 fclose(fd);
      |                 ^~~~~~~~~~
events/libdmraid-events-isw.c:171:20: note: returned from 'popen'
  171 |         FILE *fd = popen("which sgpio", "r");
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~
../lib/libdmraid.a ../lib/libdmraid.so ../lib/libdmraid-events-isw.so ../lib/libdmraid.a ../lib/libdmraid.so ../lib/libdmraid-events-isw.so
make[1]: Leaving directory '/build/dmraid/1.0.0.rc16/lib'
make -C man
make[1]: Entering directory '/build/dmraid/1.0.0.rc16/man'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/build/dmraid/1.0.0.rc16/man'
make -C tools
make[1]: Entering directory '/build/dmraid/1.0.0.rc16/tools'
gcc -MM -MF dmraid.d -I  -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE dmraid.c; \
gcc -c -I  -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE dmraid.c -o dmraid.o
gcc -MM -MF commands.d -I  -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE commands.c; \
gcc -c -I  -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE commands.c -o commands.o
gcc -MM -MF toollib.d -I  -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE toollib.c; \
gcc -c -I  -I. -I../include -I../lib -O2 -DDMRAID_NATIVE_LOG -DHAVE_GETOPTLONG -fPIC -Wall -Wundef -Wcast-align -Wwrite-strings -Winline -DDMRAID_TEST -DDMRAID_AUTOREGISTER -O2 -D_LARGEFILE64_SOURCE toollib.c -o toollib.o
gcc -o dmraid dmraid.o commands.o toollib.o -rdynamic -L../lib \
      -L/nix/store/75k5rywwbb24z7g463cixdh4qnxisjc0-dmraid-1.0.0.rc16/lib -ldmraid -ldevmapper -ldevmapper-event -ldmraid -ldevmapper -ldl 
gcc -o dmevent_tool dmevent_tool.c -I  -I. -I../include -I../lib -rdynamic -L../lib \
      -L/nix/store/75k5rywwbb24z7g463cixdh4qnxisjc0-dmraid-1.0.0.rc16/lib -ldmraid -ldevmapper -ldevmapper-event -ldmraid -ldevmapper -ldl 
dmevent_tool.c:126:8: error: return type defaults to 'int' []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-int-Wimplicit-int]8;;]
  126 | static _process_opt(int opt, const char *cmd)
      |        ^~~~~~~~~~~~
dmevent_tool.c: In function '_process_opt':
dmevent_tool.c:151:25: error: 'return' with no value, in function returning non-void []8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-return-mismatch-Wreturn-mismatch]8;;]
  151 |                         return;
      |                         ^~~~~~
dmevent_tool.c:126:8: note: declared here
  126 | static _process_opt(int opt, const char *cmd)
      |        ^~~~~~~~~~~~
make[1]: *** [Makefile:78: dmevent_tool] Error 1
make[1]: Leaving directory '/build/dmraid/1.0.0.rc16/tools'
make: *** [make.tmpl:115: tools] Error 2
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.2-zen1, NixOS, 25.05 (Warbler), 25.05.20241219.d70bd19`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.92.0-dev-pre20241211-92ed9fe
System type: x86_64-linux
Additional system types: aarch64-linux, i686-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/perchun/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/perchun/.nix-profile/etc/xdg/nix/nix.conf:/home/perchun/.local/state/nix/profile/etc/xdg/nix/nix.conf:/home/perchun/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/perchun/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/hd8gdx3wk23c3vdql1091bbm7wcrhfhy-lix-2.92.0-dev-pre20241211-92ed9fe/share`
 - nixpkgs: `/etc/nix/path/nixpkgs`
```

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@7c6f434c

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/dmraid/default.nix
+++ b/pkgs/os-specific/linux/dmraid/default.nix
@@ -16,7 +16,10 @@ stdenv.mkDerivation rec {
   };
 
   patches =
-    [ ./hardening-format.patch ]
+    [
+      ./hardening-format.patch
+      ./fix-dmevent_tool.patch
+    ]
     ++ lib.optionals stdenv.hostPlatform.isMusl [
       (fetchpatch {
         url = "https://raw.githubusercontent.com/void-linux/void-packages/fceed4b8e96b3c1da07babf6f67b6ed1588a28b2/srcpkgs/dmraid/patches/006-musl-libc.patch";

--- a/pkgs/os-specific/linux/dmraid/fix-dmevent_tool.patch
+++ b/pkgs/os-specific/linux/dmraid/fix-dmevent_tool.patch
@@ -1,0 +1,20 @@
+--- a/1.0.0.rc16/tools/dmevent_tool.c.old	2024-12-26 18:39:04.294039876 -0700
++++ a/1.0.0.rc16/tools/dmevent_tool.c	2024-12-26 18:45:59.416784807 -0700
+@@ -123,7 +123,7 @@
+ }
+ 
+ /* Increment option counters. */
+-static _process_opt(int opt, const char *cmd)
++static int _process_opt(int opt, const char *cmd)
+ {
+ 	struct opt_def_struct {
+ 		const char opt;		/* Option character. */
+@@ -148,7 +148,7 @@
+ 
+ 			optc[o->type]++;
+ 			optc[OPT_SUM]++;
+-			return;
++			return 0;
+ 		}
+ 	}
+ 

